app/javascript/controllers/home_search_controller.js

```diff
@@ -2,8 +2,118 @@ import { Controller } from "@hotwired/stimulus"
 
 // Connects to data-controller="home-search"
 export default class extends Controller {
+  static targets = [ "input", "dropDown", "ontology", "searchedOntologies", "searchOntologyContent", "homeSearchOntologies" ]
+  static values = {
+    ontologies: Array
+  }
   connect() {
+    this.ontologies = this.ontologiesValue
+    this.input = this.inputTarget
+    this.dropDown = this.dropDownTarget
+    this.ontology = this.ontologyTarget
+    this.searchedOntologies = this.searchedOntologiesTarget
+    this.searchOntologyContent = this.searchOntologyContentTarget
+    this.homeSearchOntologies = this.homeSearchOntologiesTarget
+    
+    this.input.addEventListener("input",()=>{
```

Comment:
Use stimulus actions instead of `addEventListener`.

Or you need to remove them when the controller is destroyed.   see https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener